ml-agents-envs/mlagents_envs/registry/unity_env_registry.py

```diff
@@ -0,0 +1,61 @@
+from typing import Dict, Iterator, Any
+from collections.abc import Mapping
+from mlagents_envs.registry.base_registry_entry import BaseRegistryEntry
+
+
+class UnityEnvRegistry(Mapping):
+    """
+    ### UnityEnvRegistry
+    Provides a library of Unity environments that can be launched without the need
+    of downloading the Unity Editor.
+    The UnityEnvRegistry implements a Map, to access an entry of the Registry, use:
+    ```python
+    registry = UnityEnvRegistry()
+    entry = registry[<environment_identifyier>]
+    ```
+    An entry has the following properties :
+     * `identifier` : Uniquely identifies this environment
+     * `expected_reward` : Corresponds to the reward an agent must obtained for the task
+     to be considered completed.
+     * `description` : A human readable description of the environment.
+
+    To launch a Unity environment from a registry entry, use the `make` method:
+    ```python
+    registry = UnityEnvRegistry()
+    env = registry[<environment_identifyier>].make()
+    ```
+    """
+
+    _REGISTERED_ENVS: Dict[str, BaseRegistryEntry] = {}
```

Comment:
I think this should be an instance member, not a class member. If you've got two sources for your environments, no reason not create two registries.